### DISCUSSION TITLE
Demo debugging with Xcode and scikit-build-core

### DISCUSSION THREAD

**Discussion by ianhbell** [wjakob/nanobind#586] [Q&A]
I just figured out how to debug my nanobind extension in Xcode, it was incredibly helpful to be able to. I think others would benefit from my experience, where could I write up some docs on this?

**Comment by ianhbell** [comment]
It is pretty easy to setup once you know a few tricks with the ``pyproject.toml`` and some hacking in the ``CMakeLists.txt``

**Comment by nurpax** [comment]
Is this applicable to other IDEs too?  For example, I'd be interested to know how to do this with VSCode.

**Comment by ianhbell** [comment]
Actually I tried (really hard) to get it working in VSCode, but I was unable to, and it wasn't trivial in Xcode either but I figured it out after a few tricks that were not at all obvious.

**Comment by adamryczkowski** [comment]
@ianhbell  Did you write the docs? I am interested on doing the same with CLion.

**Comment by ianhbell** [comment]
My example for CMake looks like:
```cmake
if (XCODE_DEBUG_PYTHON)
#See https://stackoverflow.com/a/72078672 and https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#properties-on-targets
set_target_properties(_${NAME}_impl PROPERTIES
                      XCODE_GENERATE_SCHEME TRUE
                      XCODE_SCHEME_WORKING_DIRECTORY "/Users/ihb/Documents/PapersInProgress/Working/(J)phasedet pure/figs"
                      XCODE_SCHEME_ARGUMENTS teqpflsh_Flasher.py
                      XCODE_SCHEME_EXECUTABLE /Users/ihb/mambaforge/envs/py310/bin/python)
endif()
```

I have no idea whether something similar is possible with CLion.

**Reply by ianhbell** [reply]
I hardcoded a lot of things that you could probably get from arguments, but it worked, which was good enough for me.
